src/pages/GapsPatternsPage.tsx

```diff
@@ -56,15 +56,22 @@ const CustomTooltip = ({ active, payload }: TooltipProps<number, string>) => {
 
 function GapsByHour({ lineRef, operatorRef, fromDate, toDate }: BusLineStatisticsProps) {
   const [sortingMode, setSortingMode] = useState<'hour' | 'severity'>('hour')
-
+  const [isLoadingData, setIsLoadingData] = useState(true)
   const hourlyData = useGapsList(fromDate, toDate, operatorRef, lineRef, sortingMode)
```

Comment:
```suggestion
  const hourlyData = useGapsList(fromDate, toDate, operatorRef, lineRef, sortingMode)
  const isLoading = hourlyData.length
```

Reply:
Much cleaner, fixed that too!
Changed the variable name to `isLoading` instead of `isLoadingData` and fixed a minor mistake in your change:
`const isLoading = hourlyData.length` to `const isLoading = !hourlyData.length`.
Thanks for the review 👍